libraries/AP_Scripting/applets/quadplane_terrain_avoid.lua

```diff
@@ -0,0 +1,1331 @@
+
+--[[
+    // @Param: TA_CMTC_HGT
+    // @DisplayName: CMTC Height
+    // @Description: The minimum Height above terrain to maintain when following an AUTO mission or RTL. If zero(0) use TA_PTCH_DOW_MIN.
+    // @Units: m
+--]]
+TA_CMTC_HGT = bind_add_param("CMTC_HGT", 12, 50)
+
+--[[
+    // @Param: TA_CMTC_ENABLE
+    // @DisplayName: CMTC Enable
+    // @Description: Whether to enable Can't Make That Climb while running Terrain Avoidance
+    // @Range: 0 1
+--]]
+TA_CMTC_ENABLE = bind_add_param("CMTC_ENABLE", 13, 0)
+
+--[[
+    // @Param: TA_UPDATE_RATE
+    // @DisplayName: Frequency to process avoidance
+    // @Description: Avoidance processing rate
+    // @Units: Hz
+--]]
+TA_UPDATE_RATE = bind_add_param("UPDATE_RATE", 14, 10)
+
+--[[
+    // @Param: TA_CMTC_RAD
+    // @DisplayName: CMTC loiter radius
+    // @Description: Use this radius for the loiter when trying to gain altitude. If not set or <=0 use WP_LOITER_RAD
+    // @Units: m
+--]]
+TA_CMTC_RAD = bind_add_param("CMTC_RAD", 15, 0)
+
+local pitch_down_min = TA_PTCH_DWN_MIN:get()
+local pitch_forward_min = TA_PTCH_FWD_MIN:get()
+local pitch_timeout = TA_PTCH_TIMEOUT:get()
+local home_distance_max = TA_HOME_DIST:get() or 75
+
+local quad_down_min = TA_QUAD_DWN_MIN:get()
+local quad_forward_min = TA_QUAD_FWD_MIN:get()
+
+local altitude_max = TA_ALT_MAX:get()
+
+local groundspeed_max = TA_GSP_MAX:get() or -1
+local groundspeed_airbrake_limit = TA_GSP_AIRBRAKE:get() or 0
+
+local cmtc_height_m = TA_CMTC_HGT:get()
+if cmtc_height_m == 0 then
+    cmtc_height_m = pitch_down_min
+end
+local cmtc_enable = TA_CMTC_ENABLE:get() or 0
+
+MIN_ALT_MAX = 20
+
+REFRESH_RATE = 1.0 / (TA_UPDATE_RATE:get() or 10)
+
+local rangefinder_down_value = 0.0
+local rangefinder_forward_value = 0.0
+MAX_RANGEFINDER_VALUE = 90
+
+Q_ENABLE = Parameter('Q_ENABLE')
+Q_WVANE_ENABLE = Parameter('Q_WVANE_ENABLE')
+Q_TILT_ENABLE = Parameter('Q_TILT_ENABLE')
+THR_MAX = Parameter('THR_MAX')
+AIRSPEED_MIN = Parameter("AIRSPEED_MIN")
+AIRSPEED_CRUISE = Parameter("AIRSPEED_CRUISE")
+AIRSPEED_MAX = Parameter("AIRSPEED_MAX")
+WP_LOITER_RAD = Parameter("WP_LOITER_RAD")
+TERRAIN_ENABLE = Parameter("TERRAIN_ENABLE")
+TERRAIN_SPACING = Parameter("TERRAIN_SPACING")
+PTCH_LIM_MAX_DEG = Parameter("PTCH_LIM_MAX_DEG")
+TECS_CLMB_MAX = Parameter("TECS_CLMB_MAX")
+local airspeed_min = AIRSPEED_MIN:get() or 10
+local airspeed_cruise = AIRSPEED_CRUISE:get() or 18
+local airspeed_max = AIRSPEED_MAX:get() or 30
+local wp_loiter_rad_m = WP_LOITER_RAD:get() or 150
+local cmtc_rad_m = TA_CMTC_RAD:get() or wp_loiter_rad_m
+local terrain_spacing = TERRAIN_SPACING:get() or 100
+local ptch_lim_max_deg = PTCH_LIM_MAX_DEG:get() or 25
+local tecs_climb_max = TECS_CLMB_MAX:get() or 5.0
+
+if cmtc_rad_m <= 0 then
+    cmtc_rad_m = wp_loiter_rad_m
+end
+
+THROTTLE_CHANNEL = rc:get_channel(Parameter('RCMAP_THROTTLE'):get()) -- The RC channel used for throttle
+THROTTLE_UP_PWM = 1800
+THROTTLE_HOVER_PWM = 1500
```

Comment:
Better to pull this out from params. 1500 may not be hover on all vehicles. It would be the average of RC min and max.